### DISCUSSION TITLE
Confusion matrix axis

### DISCUSSION THREAD

**Discussion by mfoglio** [Lightning-AI/pytorch-lightning#13248] [General]
Hello, 
I am computing a confusion matrix using torch lightning and torch metrics. What are the axis used by the library?

Which of the following two options is used?
- Option 1: y-axis label is `true (aka actual)`; a-axis label is `predicted` https://scikit-learn.org/0.20/_images/sphx_glr_plot_confusion_matrix_001.png )
- Option 2: y-axis label is `predicted`; x-axis label is `true (aka actual)` (example https://i.stack.imgur.com/xazfA.png )

I think Option 1 is much more common. However, I'd like for anyone to confirm that this is the case for torch metrics. 
Thank you!

**Comment by mfoglio** [comment]
Hello, is there anyone who knows the answer?

**Reply by jowagner** [reply]
You can work it out by printing the confusion matrix for your test or validation set with some classifier, calculating the row and column totals and comparing these numbers to the label frequencies in the test or validation set. In the unlikely event that both types of totals match, train another classifier. Accuracy of the classifier does not matter. Even a freshly initialised model will do for this purpose.
